app/src/main/java/app/revanced/integrations/patches/components/LithoFilterPatch.java

```diff
@@ -322,42 +310,90 @@ protected ByteTrieSearch createSearchGraph() {
+        IDENTIFIER,
+        PATH,
+        PROTOBUFFER
+    }
+
     /**
-     * All group filters must be set before the constructor call completes.
-     * Otherwise {@link #isFiltered(String, String, byte[], FilterGroupList, FilterGroup, int)}
-     * will never be called for any matches.
+     * Identifier callbacks.  Do not add to this instance,
+     * and instead use {@link #addIdentifierCallbacks(StringFilterGroup...)}.
      */
+    protected final List<StringFilterGroup> identifierCallbacks = new ArrayList<>();
+    /**
+     * Path callbacks. Do not add to this instance,
+     * and instead use {@link #addPathCallbacks(StringFilterGroup...)}.
+     */
+    protected final List<StringFilterGroup> pathCallbacks = new ArrayList<>();
 
-    protected final StringFilterGroupList pathFilterGroupList = new StringFilterGroupList();
-    protected final StringFilterGroupList identifierFilterGroupList = new StringFilterGroupList();
+    /**
+     * Adds callbacks to {@link #isFiltered(String, String, byte[], StringFilterGroup, FilterContentType, int)}
+     * if any of the groups are found.
+     */
+    protected final void addIdentifierCallbacks(StringFilterGroup... groups) {
+        identifierCallbacks.addAll(Arrays.asList(groups));
+    }
+
+    /**
+     * Adds callbacks to {@link #isFiltered(String, String, byte[], StringFilterGroup, FilterContentType, int)}
+     * if any of the groups are found.
+     */
+    protected final void addPathCallbacks(StringFilterGroup... groups) {
+        pathCallbacks.addAll(Arrays.asList(groups));
+    }
 
     /**
      * Called after an enabled filter has been matched.
-     * Default implementation is to always filter the matched item.
+     * Default implementation is to always filter the matched component and log the action.
      * Subclasses can perform additional or different checks if needed.
      * <p>
+     * If the content is to be filtered, subclasses should always
+     * call this method (and never return a plain 'true').
+     * That way the logs will always show when a component was filtered and which filter hide it.
+     * <p>
      * Method is called off the main thread.
      *
-     * @param matchedList  The list the group filter belongs to.
      * @param matchedGroup The actual filter that matched.
-     * @param matchedIndex Matched index of string/array.
-     * @return True if the litho item should be filtered out.
+     * @param contentType  The type of content matched.
+     * @param contentIndex Matched index of the identifier or path.
+     * @return True if the litho component should be filtered out.
      */
-    @SuppressWarnings("rawtypes")
     boolean isFiltered(@Nullable String identifier, String path, byte[] protobufBufferArray,
-                       FilterGroupList matchedList, FilterGroup matchedGroup, int matchedIndex) {
+                       StringFilterGroup matchedGroup, FilterContentType contentType, int contentIndex) {
```

Comment:
To reduce boilerplate, it might be a good idea to create a FilterResult. This could hold state on the `contentType` the `matchedGroup`, `contentIntex` and similar so the signature of the method `isFiltered` is reduced. What do you think?